src/components/Buildbaron/BBFIleTicket.tsx

```diff
@@ -0,0 +1,151 @@
+import { BBTitle, TitleAndButtons } from "./BBComponents";
+import { BuildBaronTable } from "./BuildBaronTable";
+
+export const BBFileTicket: React.FC<{
+  taskId: string;
+  dispatchBanner;
+  setCreatedTicketsCount;
+  createdTicketsCount;
+}> = ({
+  taskId,
+  dispatchBanner,
+  setCreatedTicketsCount,
+  createdTicketsCount,
+}) => (
+  <>
+    <BBTitle margin>Create a new ticket in Jira </BBTitle>
+    <FileTicket
+      taskId={taskId}
+      dispatchBanner={dispatchBanner}
+      setCreatedTicketsCount={setCreatedTicketsCount}
+      createdTicketsCount={createdTicketsCount}
+    />
+  </>
+);
+
+interface FileTicketProps {
+  taskId: string;
+  dispatchBanner;
+  setCreatedTicketsCount;
+  createdTicketsCount: number;
+}
+
+export const FileTicket: React.FC<FileTicketProps> = ({
+  taskId,
+  dispatchBanner,
+  setCreatedTicketsCount,
+  createdTicketsCount,
+}) => {
+  const [fileJiraTicket, { loading: loadingFileJiraTicket }] = useMutation<
+    BbCreateTicketMutation,
+    BbCreateTicketMutationVariables
+  >(FILE_JIRA_TICKET, {
+    onCompleted: () => {
+      setButtonText("FILE ANOTHER TICKET");
+      dispatchBanner.successBanner(
+        `Ticket successfully created for this task.`
+      );
+      setCreatedTicketsCount(createdTicketsCount + 1);
+    },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error filing the ticket: ${error.message}`
+      );
+    },
+  });
+
+  const [buttonText, setButtonText] = useState<string>("FILE TICKET");
+  const onClickFile = () => {
+    // todo: add analytics
+    fileJiraTicket({ variables: { taskId } });
+  };
+
+  return (
+    <Popconfirm
+      title="Do you want to create a failure ticket for this task?"
+      onConfirm={onClickFile}
+      icon={null}
+      placement="right"
+      okText="File Ticket"
+      okButtonProps={{ loading: loadingFileJiraTicket }}
+      cancelButtonProps={{ disabled: loadingFileJiraTicket }}
+    >
+      <FileButton
+        dataCy="file-ticket-button"
+        variant={Variant.Primary}
+        size="small"
+      >
+        {buttonText}
+      </FileButton>
+    </Popconfirm>
+  );
+};
+
+interface CreatedTicketsProps {
+  taskId: string;
+  dispatchBanner;
+  setCreatedTicketsCount;
+  createdTicketsCount;
+}
+
+export const CreatedTickets: React.FC<CreatedTicketsProps> = ({
+  taskId,
+  dispatchBanner,
+  setCreatedTicketsCount,
+  createdTicketsCount,
+}) => {
+  const { data, startPolling, stopPolling } = useQuery<
+    GetCreatedTicketsQuery,
+    GetCreatedTicketsQueryVariables
+  >(GET_CREATED_TICKETS, {
+    variables: { taskId },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error getting tickets created for this task: ${error.message}`
+      );
+    },
+  });
+  const length = data?.bbGetCreatedTickets?.length;
+
+  if (createdTicketsCount > data?.bbGetCreatedTickets?.length) {
+    startPolling(1 * 1000);
```

Comment:
We have a constant for polling https://github.com/evergreen-ci/spruce/blob/8bed0a6195d619632f5d4277a52e47639ce675c1/src/constants/index.ts#L1

Reply:
the constant for polling is too large for me. I want to poll one second after the the ticket is filed, and stop polling once the new ticket shows up.